Neos.Flow/Classes/Core/Booting/Scripts.php

```diff
@@ -687,13 +687,9 @@ protected static function buildSubprocessCommand($commandIdentifier, array $sett
         }
 
         $escapedArguments = '';
-        if ($commandArguments !== []) {
-            foreach ($commandArguments as $argument => $argumentValue) {
-                $escapedArguments .= ' ' . escapeshellarg('--' . trim($argument));
-                if (trim($argumentValue) !== '') {
-                    $escapedArguments .= ' ' . escapeshellarg(trim($argumentValue));
-                }
-            }
+        foreach ($commandArguments as $argument => $argumentValue) {
+            $argumentValue = trim($argumentValue);
+            $escapedArguments .= ' --' . trim($argument) . ($argumentValue !== '' ? '=' . escapeshellarg($argumentValue) : '');
```

Comment:
With that `escapeshellarg` is no longer applied to the argument name. That might be a problem, no?

Reply:
See commit message. An argument name can neither contain spaces nor special characters-

Reply:
> IMHO the argument name cannot and should not be something that needs escaping

But effectively this would introduce a potential vulnerability if $someone forwards the argument from an untrusted source, wouldn't it?

Reply:
In theory, yes, but we are looking at a low level method here, if you forward an argument name from an untrusted source to a command line command you made mistakes already.
But if you prefer I can reinstate that :)

Reply:
I'd leave it in place, TBH.

Reply:
I don't want to split hairs, but it's called by highlevel (API) methods (`executeCommand(Async)()`) and the commandArguments are passed through without validation.